app/models/user.rb

```diff
@@ -79,4 +81,20 @@ def create_api_token
     token.save!
     token
   end
+
+  def active_for_authentication?
+    super && self.is_active?
+  end
+
+  def inactive_message
+    self.is_active? ? super : I18n.t('devise.failure.suspended')
+  end
+
+  scope :within, -> (institution_or_site) {
+    if institution_or_site.is_a?(Institution)
+      where(institution: institution_or_site)
+    else
+      where("prefix LIKE concat(?, '%')", institution_or_site.prefix)
+    end
+  }
```

Comment:
This scope is failing for both institutions and sites. Note that user does not directly have an institution or prefix fields, this should probably be resolved via the `role_users` relationship, checking the associated site or institution of all the related roles.